src/imports.rs

```diff
@@ -0,0 +1,766 @@
+            let err = format!("Package '{src}' does not contain the top level name '{nam}'");
+            diag.add_book_error(err);
+          }
+
+          let book = self.get_book_mut(idx);
+
+          let src = format!("{}/{}", src, nam);
+          let aliased = alias.unwrap_or(nam);
+          book.import_ctx.map.add_bind(aliased, &src, diag);
+        }
+
+        (BoundSource::File(src), ImportType::List(names)) => {
+          let bound_book = self.books.get(&src).unwrap();
+
+          for (sub, _) in &names {
+            if !bound_book.top_level_names().contains(&sub) {
+              let err = format!("Package '{src}' does not contain the top level name '{sub}'");
+              diag.add_book_error(err);
+            }
+          }
+
+          let book = self.get_book_mut(idx);
+
+          for (sub, alias) in names {
+            let src = format!("{}/{}", src, sub);
+            let aliased = alias.unwrap_or(sub);
+            book.import_ctx.map.add_bind(aliased, &src, diag);
+          }
+        }
+
+        (BoundSource::File(src), ImportType::Glob) => {
+          let bound_book = self.books.get(&src).unwrap();
+          let names: HashSet<_> = bound_book.top_level_names().cloned().collect();
+
+          let book = self.get_book_mut(idx);
+
+          for sub in names {
+            let src = format!("{}/{}", src, sub);
+            book.import_ctx.map.add_bind(sub, &src, diag);
+          }
+        }
+
+        (BoundSource::Folder(mut src), ImportType::Simple(nam, alias)) => {
+          let src = src.pop().unwrap();
+          self.add_book_bind(idx, src, nam, alias, diag);
+        }
+
+        (BoundSource::Folder(pkgs), ImportType::List(names)) => {
+          for (src, (nam, alias)) in pkgs.into_iter().zip_eq(names) {
+            self.add_book_bind(idx, src, nam, alias, diag);
+          }
+        }
+
+        (BoundSource::Folder(pkgs), ImportType::Glob) => {
+          for src in pkgs {
+            let nam = Name::new(src.split('/').last().unwrap());
+            self.add_book_bind(idx, src, nam, None, diag);
+          }
+        }
+
+        (BoundSource::None, _) => unreachable!(),
+      }
+    }
+  }
+
+  fn add_book_bind(&mut self, idx: usize, src: Name, nam: Name, alias: Option<Name>, diag: &mut Diagnostics) {
+    let bound_book = self.books.get(&src).unwrap();
+    let names: HashSet<_> = bound_book.top_level_names().cloned().collect();
+
+    let aliased = alias.as_ref().unwrap_or(&nam);
+
+    let book = self.get_book_mut(idx);
+
+    for name in &names {
+      if name != &nam {
+        let src = format!("{}/{}", src, name);
+        let bind = Name::new(format!("{aliased}/{name}"));
+        book.import_ctx.map.add_bind(bind, &src, diag);
+      }
+    }
+
+    if names.contains(&nam) {
+      let src = format!("{}/{}", src, nam);
+      book.import_ctx.map.add_bind(aliased.clone(), &src, diag);
+    }
+  }
+}
+
+impl ParseBook {
+  pub fn load_imports(
```

Comment:
Can you add a comment here or somewhere about the general strategy/logic of the import finder/loader?

Reply:
Nwo that you separated it into different submodules, maybe this explanation should go as documentation of the `imports` module.

Reply:
Like at the top of the imports/mod.rs?